cardano-api/test/Test/Cardano/Api/TxBody.hs

```diff
@@ -0,0 +1,221 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Test.Cardano.Api.TxBody (tests) where
+
+import           Cardano.Prelude
+
+import           Hedgehog (Property, forAll, property, tripping)
+import           Test.Tasty (TestTree)
+import           Test.Tasty.Hedgehog (testProperty)
+import           Test.Tasty.TH (testGroupGenerator)
+
+import           Cardano.Api
+
+import           Test.Cardano.Api.Typed.Gen (genTxBody, genTxBodyContent)
+
+
+prop_roundtrip_TxBody_make_get_Byron :: Property
+prop_roundtrip_TxBody_make_get_Byron = roundtripTxBodyMakeGet ByronEra
+
+prop_roundtrip_TxBody_make_get_Shelley :: Property
+prop_roundtrip_TxBody_make_get_Shelley = roundtripTxBodyMakeGet ShelleyEra
+
+prop_roundtrip_TxBody_make_get_Allegra :: Property
+prop_roundtrip_TxBody_make_get_Allegra = roundtripTxBodyMakeGet AllegraEra
+
+prop_roundtrip_TxBody_make_get_Mary :: Property
+prop_roundtrip_TxBody_make_get_Mary = roundtripTxBodyMakeGet MaryEra
+
+-- TODO Alonzo
+-- prop_roundtrip_TxBody_make_get_Alonzo :: Property
+-- prop_roundtrip_TxBody_make_get_Alonzo = roundtripTxBodyMakeGet AlonzoEra
+
+
+roundtripTxBodyMakeGet :: IsCardanoEra era => CardanoEra era -> Property
+roundtripTxBodyMakeGet era =
+  property $ do
+    content <- forAll $ genTxBodyContent era
+    tripping
+      (normalizeOriginal $ review content)
+      (\_ -> makeTransactionBody content)
+      (<&> \case
+        TxBody content' -> normalizeRoundtrip content'
+        _               -> panic "impossible")
+
+
+normalizeOriginal :: TxBodyContent ViewTx era -> TxBodyContent ViewTx era
+normalizeOriginal content =
+  content
+    { txIns          = sortOn fst            $ txIns          content
+    , txMetadata     = normalizeMetadata     $ txMetadata     content
+    , txAuxScripts   = normalizeAuxScripts   $ txAuxScripts   content
+    , txWithdrawals  = normalizeWithdrawals  $ txWithdrawals  content
+    , txCertificates = normalizeCertificates $ txCertificates content
+    , txMintValue    = normalizeMintValue    $ txMintValue    content
+    }
```

Comment:
Is it documented anywhere what it means to "normalize"?

Reply:
Added a line before. Is it OK or needs more docs?